File: php/src/Media/MediaModelsService.php
<?php

declare(strict_types=1);

namespace Buble\Media;

use Buble\Http\TransportInterface;
use Buble\RequestOptions;

final class MediaModelsService
{
    public function __construct(private readonly TransportInterface $transport)
    {
    }

    /**
     * @return array<string, mixed>
     */
    public function list(?string $mediaType = null): array
    {
        $query = [];
        if ($mediaType !== null && $mediaType !== '') {
            $query['media_type'] = $mediaType;
        }

        return $this->transport->request('GET', '/api/v1/media_models', options: new RequestOptions(query: $query));
    }
}

File: dotnet/src/Buble.Sdk/Media/MediaModelsService.cs
using Buble.Sdk.Http;

namespace Buble.Sdk.Media;

public sealed class MediaModelsService
{
    private readonly BubleHttpClient _http;

    internal MediaModelsService(BubleHttpClient http)
    {
        _http = http;
    }

    public Task<Envelope<IReadOnlyList<MediaModel>>?> ListAsync(
        string? mediaType = null,
        CancellationToken cancellationToken = default)
    {
        var options = new RequestOptions();
        if (!string.IsNullOrWhiteSpace(mediaType))
        {
            options.Query["media_type"] = mediaType!;
        }

        return _http.GetAsync<Envelope<IReadOnlyList<MediaModel>>>("/api/v1/media_models", options, cancellationToken);
    }
}
